internal/dataplane/parser/ingressclassparemeters_test.go

```diff
@@ -0,0 +1,158 @@
+		parameterSpec *configurationv1alpha1.IngressClassParametersSpec
+		hasError      bool
+	}{
+		{
+			name:          "nil-paramref",
+			parameterSpec: defaultIcpSpec,
+			hasError:      true,
+		},
+		{
+			name: "nil-apigroup",
+			paramRef: &netv1.IngressClassParametersReference{
+				Kind: "configMap",
+				Name: "some-cm",
+			},
+			parameterSpec: defaultIcpSpec,
+			hasError:      true,
+		},
+		{
+			name: "nil-scope",
+			paramRef: &netv1.IngressClassParametersReference{
+				APIGroup:  &configurationv1alpha1.GroupVersion.Group,
+				Kind:      configurationv1alpha1.IngressClassParametersKind,
+				Namespace: &testNamespaceName,
+				Name:      testIcpName,
+			},
+			parameterSpec: defaultIcpSpec,
+			hasError:      true,
+		},
+		{
+			name: "nil-namespace",
+			paramRef: &netv1.IngressClassParametersReference{
+				APIGroup: &configurationv1alpha1.GroupVersion.Group,
+				Kind:     configurationv1alpha1.IngressClassParametersKind,
+				Scope:    &scopeNamespace,
+				Name:     testIcpName,
+			},
+			parameterSpec: defaultIcpSpec,
+			hasError:      true,
+		},
+		{
+			name: "matched-parameters",
+			paramRef: &netv1.IngressClassParametersReference{
+				APIGroup:  &configurationv1alpha1.GroupVersion.Group,
+				Kind:      configurationv1alpha1.IngressClassParametersKind,
+				Scope:     &scopeNamespace,
+				Namespace: &testNamespaceName,
+				Name:      testIcpName,
+			},
+			parameterSpec: &icp.Spec,
+		},
+		{
+			name: "unmatched-kind",
+			paramRef: &netv1.IngressClassParametersReference{
+				APIGroup:  &configurationv1alpha1.GroupVersion.Group,
+				Kind:      "SomeKind",
+				Scope:     &scopeNamespace,
+				Namespace: &testNamespaceName,
+				Name:      testIcpName,
+			},
+			parameterSpec: defaultIcpSpec,
+			hasError:      true,
+		},
+		{
+			name: "unmatched-namespace",
+			paramRef: &netv1.IngressClassParametersReference{
+				APIGroup:  &configurationv1alpha1.GroupVersion.Group,
+				Kind:      configurationv1alpha1.IngressClassParametersKind,
+				Scope:     &scopeNamespace,
+				Namespace: new(string),
+				Name:      testIcpName,
+			},
+			parameterSpec: defaultIcpSpec,
+			hasError:      true,
+		},
+		{
+			name: "unmatched-name",
+			paramRef: &netv1.IngressClassParametersReference{
+				APIGroup:  &configurationv1alpha1.GroupVersion.Group,
+				Kind:      configurationv1alpha1.IngressClassParametersKind,
+				Scope:     &scopeNamespace,
+				Namespace: &testNamespaceName,
+				Name:      "another-icp",
+			},
+			parameterSpec: defaultIcpSpec,
+			hasError:      true,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			ingressClass := &netv1.IngressClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: testNamespaceName,
+					Name:      tc.name,
+				},
+				Spec: netv1.IngressClassSpec{
+					Parameters: tc.paramRef,
+				},
+			}
+			cacheStores, err := store.NewCacheStoresFromObjs()
+			assert.NoError(t, err)
```

Comment:
Those would better use `require` instead to prevent continuing execution, so that when an error occurs we don't continue the test as it would be pointless.